examples/src/main/java/com/teamdev/jxbrowser/examples/PrintSettings.java

```diff
@@ -42,43 +42,44 @@
-            PrintJob<HtmlSettings> printJob = printer.printJob();
-            printJob.settings()
-                    .header("<span style='font-size: 12px;'>Page header:</span>"
-                            + "<span class='title'></span>")
-                    .footer("<span style='font-size: 12px;'>Page footer:</span>"
-                            + "<span class='pageNumber'></span>")
-                    .paperSize(ISO_A4)
-                    .colorModel(COLOR)
-                    .enablePrintingBackgrounds()
-                    .disablePrintingHeaderFooter()
-                    .orientation(PORTRAIT)
-                    .apply();
-            // #enddocfragment "Configure settings"
-            // #docfragment "Subscribe to PrintCompleted"
-            printJob.on(PrintCompleted.class, event -> {
-                if (event.isSuccess()) {
-                    System.out.println("Printing is completed successfully.");
-                } else {
-                    System.out.println("Printing has failed.");
-                }
+        try (Engine engine = Engine.newInstance(HARDWARE_ACCELERATED)) {
+            Browser browser = engine.newBrowser();
+            browser.set(PrintCallback.class, (params, tell) -> tell.print());
+            // #docfragment "Callback"
+            browser.set(PrintHtmlCallback.class, (params, tell) -> {
+                // #docfragment "Configure settings"
+                SystemPrinter<HtmlSettings> printer = params.printers()
+                        .defaultPrinter()
+                        .orElseThrow(IllegalStateException::new);
+                PrintJob<HtmlSettings> printJob = printer.printJob();
+                printJob.settings()
+                        .header("<span style='font-size: 12px;'>Page header:</span>"
+                                + "<span class='title'></span>")
+                        .footer("<span style='font-size: 12px;'>Page footer:</span>"
+                                + "<span class='pageNumber'></span>")
+                        .paperSize(ISO_A4)
+                        .colorModel(COLOR)
+                        .enablePrintingBackgrounds()
+                        .disablePrintingHeaderFooter()
+                        .orientation(PORTRAIT)
+                        .apply();
+                // #enddocfragment "Configure settings"
+                // #docfragment "Subscribe to PrintCompleted"
+                printJob.on(PrintCompleted.class, event -> {
+                    if (event.isSuccess()) {
+                        System.out.println("Printing is completed successfully.");
+                    } else {
+                        System.out.println("Printing has failed.");
+                    }
+                });
+                // #enddocfragment "Subscribe to PrintCompleted"
+                // #docfragment "Proceed"
+                tell.proceed(printer);
+                // #enddocfragment "Proceed"
```

Comment:
@vladimir-ikryanov, a number of formatting suppressions used in this code is code smell which screams for refactoring.

I understand that you intend to have simpler code. That's why it's under `public static void main`. But in cases like this it obfuscates the scene. I, as a newbie user, can ho idea about the essence of the lambdas we as around. In combination with HTML code around it looks like a puzzling mess.

Reply:
Discussed with @vladimir-ikryanov. I recalled these are code fragments for documentation excerpts. Agreed to have a small app class with refactored methods. Those methods can have Javadocs explaining the nature of the operations. Also, code fragment fences would look more natural around in method bodies.